### ISSUE TITLE
URL query params should be optional

### ISSUE THREAD

**Issue by hftf** [qbreader/website#294]
For example, if any clause from the query string below is removed, then 0 results are returned. App should assume that a param is empty by default when not explicitly given. This should only really effect URL human-readability, so not a high priority, but it is nice to have for sharing search results or handling URLs manually.

https://www.qbreader.org/db?queryString=foo&categories=&subcategories=&alternateSubcategories=&difficulties=&maxReturnLength=&questionType=all&randomize=false&exactPhrase=false&powermarkOnly=false&regex=false&ignoreWordOrder=false&searchType=all&setName=&tossupPagination=1&bonusPagination=1&minYear=&maxYear=

Compare:

https://aseemsdb.me/results?query=foo

**Comment by hftf** [comment]
Never mind, I made an assumption and got it wrong, so the OP isn't quite the issue. I think the issue comes about from copying the URL altered by pushState or whatever and trying to freshly navigate to it.

**Comment by geoffrey-wu** [comment]
Yeah, I'm not really sure how to make the URL play nice with the actual results, so this is the best I got for now. I'll revisit the URL and see if I can clean it up.
